src/python/parser.py

```diff
@@ -97,6 +99,23 @@ def parse_function(self):
             ftype = self.parse_type()
         contents = self.parse_block()
         return Node(FUNCTION, [ftype, fname, ptypes, pnames, contents])
+    
+    def parse_struct(self):
+        self.consume(T.STRUCT)
+        sname = self.consume(T.IDENTIFIER)
+        contents = self.parse_struct_block()
+        return Node(STRUCT, contents)
```

Comment:
The name of the struct needs to be passed in as a part of the node's children. The children parameter should also be a list.